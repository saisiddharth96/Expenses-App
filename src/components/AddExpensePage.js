import React from "react";
import {connect} from "react-redux"
import ExpenseForm from './ExpenseForm.js'
import {addExpense} from '../actions/expenses.js'

const AddExpensePage = (props) => (
  <div>
    <p>This is from Expense Page</p>
    <ExpenseForm buttonName = "Add Expense" onSubmit = {(expense) => {
      props.dispatch(addExpense(expense));
      props.history.push("/")
    }}/>
  </div>
);

export default connect()(AddExpensePage);