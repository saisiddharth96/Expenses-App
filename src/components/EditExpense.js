import React from "react";
import ExpenseForm from "./ExpenseForm.js";
import { connect } from "react-redux";
import { removeExpense, editExpense } from "../actions/expenses.js";

const EditExpense = props => {
  // console.log(props);
  return (
    <div>
      <ExpenseForm
        expense={props.expense}
        buttonName="Edit Expense"
        onSubmit={expense => {
          props.dispatch(editExpense(props.expense.id, expense));
          props.history.push("/");
          // console.log("updated", expense);
        }}
      />
      <button
        onClick={() => {
          props.dispatch(removeExpense({ id: props.expense.id }));
          props.history.push("/");
        }}
      >
        Remove
      </button>
    </div>
  );
};

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find(expense => {
      return expense.id === props.match.params.id;
    })
  };
};

export default connect(mapStateToProps)(EditExpense);
