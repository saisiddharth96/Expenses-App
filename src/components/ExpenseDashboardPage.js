import React from 'react'


const ExpenseDashboardPage = () => (
  <div>
    <p>This is from dashboard component</p>
  </div>
);

export default ExpenseDashboardPage;