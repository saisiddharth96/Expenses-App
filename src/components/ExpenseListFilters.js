import React, { Component } from "react";
import { connect } from "react-redux";
import {
  setTextFilter,
  sortByAmount,
  sortByDate,
  setEndDate,
  setStartDate
} from "../actions/filters.js";
import { DateRangePicker } from "react-dates";
import uuid from 'uuid'

class ExpenseListFilter extends Component {
  state = {
    calenderFocussed: null,
    endDateId : uuid(),
    startDateId : uuid()
  };

  onDatesChange = ({ startDate, endDate }) => {
    this.props.dispatch(setStartDate(startDate));
    this.props.dispatch(setEndDate(endDate));
  };

  onFocusChanged = calenderFocussed => {
    this.setState(() => ({ calenderFocussed }));
  };
  render() {
    return (
      <div>
        <input
          type="text"
          value={this.props.filters.text}
          onChange={e => {
            this.props.dispatch(setTextFilter(e.target.value));
          }}
        />
        <select
          value={this.props.filters.sortBy}
          onChange={e => {
            e.target.value === "date"
              ? this.props.dispatch(sortByDate())
              : this.props.dispatch(sortByAmount());
          }}
        >
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>
        <DateRangePicker
          startDate={this.props.filters.startDate}
          endDate={this.props.filters.endDate}
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calenderFocussed}
          onFocusChange={this.onFocusChanged}
          numberOfMonths={1} 
          isOutsideRange={() => false}
          startDateId = {this.state.startDateId}
          endDateId = {this.state.endDateId}
          showClearDates = {true}
        />
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    filters: state.filters
  };
};
export default connect(mapStateToProps)(ExpenseListFilter);
