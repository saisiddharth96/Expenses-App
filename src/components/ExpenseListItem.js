import React from "react";
import { NavLink } from "react-router-dom";

export const ExpenseListItem = ({ id, description, amount, createdAt }) => (
  <div>
    <h3>
      <NavLink to={"edit/" + id} activeClassName="is-active" exact={true}>
        {description}
      </NavLink>
    </h3>
    <h2>{amount}</h2>
    <h3>{createdAt}</h3>
  </div>
);

export default ExpenseListItem;
