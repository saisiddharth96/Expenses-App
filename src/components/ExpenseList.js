import React from "react";
import { connect } from "react-redux";

import ExpenseListItem from "./ExpenseListItem.js";
import getVisibleExpenses from "../selectors/expenses.js";

export const ExpenseList = props => (
  <div>
    <h1>Expense List component</h1>
    {props.expenses.length === 0 ? (
      <p>No expenses yet! Please add some</p>
    ) : (
      props.expenses.map(expense => (
        <ExpenseListItem
          key={expense.id}
          description={expense.description}
          amount={expense.amount}
          createdAt={expense.createdAt}
          id={expense.id}
          dispatch
        />
      ))
    )}
  </div>
);

const mapStateToProps = state => {
  return {
    expenses: getVisibleExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpenseList);
