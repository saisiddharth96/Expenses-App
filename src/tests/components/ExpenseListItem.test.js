import React from "react";
import { shallow } from "enzyme";
import  expenses  from "../fixtures/expenses.js";
import { ExpenseListItem } from "../../components/ExpenseListItem";

const data = expenses[1];

test("should render a single", () => {
  const wrapper = shallow(
    <ExpenseListItem
      id={data.id}
      description={data.description}
      amount={data.amount}
      createdAt = {data.createdAt}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
