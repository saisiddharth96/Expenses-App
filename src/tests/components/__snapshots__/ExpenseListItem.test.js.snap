// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a single 1`] = `
<div>
  <h3>
    <NavLink
      activeClassName="is-active"
      aria-current="page"
      exact={true}
      to="edit/2"
    >
      Rent
    </NavLink>
  </h3>
  <h2>
    500
  </h2>
  <h3>
    -345600000
  </h3>
</div>
`;
